apps/challenges/views.py

```diff
@@ -796,6 +796,18 @@ def create_challenge_using_zip_file(request, challenge_host_team_pk):
             return Response(
                 response_data, status=status.HTTP_406_NOT_ACCEPTABLE
             )
+        if "is_submission_public" in data.keys() and "is_restricted_to_select_one_submission" in data.keys():
```

Comment:
Instead of checking if these keys exist, shouldn't we check their values? 
something like:

```python
if data.get('is_submission_public') and data.get('is_restricted_to_select_one_submission'):
     # do something
```